sdk/cosmos/microsoft-azure-cosmos-benchmark/src/main/java/com/azure/data/cosmos/benchmark/ReadMyWriteWorkflow.java

```diff
@@ -124,8 +132,14 @@ protected void performWorkload(BaseSubscriber<Document> baseSubscriber, long i)
         }
 
         concurrencyControlSemaphore.acquire();
+        logger.debug("concurrencyControlSemaphore: {}", concurrencyControlSemaphore);
 
-        obs.subscribeOn(Schedulers.parallel()).subscribe(baseSubscriber);
+        try {
+            obs.subscribeOn(Schedulers.parallel()).subscribe(baseSubscriber);
+        } catch (Throwable error) {
+            concurrencyControlSemaphore.release();
+            logger.error("subscription failed due to ", error);
```

Comment:
subscribe should not ever through at least on v2. Have you seen cases of `subscribe` throwing?

if so could you please share the exception stacktrace on the code review comment?

Reply:
thanks for finding this. it is now corrected.